/*
 * Copyright (C) 2018 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

package android.os.statsd;

option java_package = "com.android.os";
option java_outer_classname = "ShellConfig";

import "frameworks/base/cmds/statsd/src/statsd_config.proto";

message PulledAtomSubscription {
    optional SimpleAtomMatcher matcher = 1;

    /* gap between two pulls in milliseconds */
    optional int32 freq_millis = 2;

    /* Packages that the pull is requested from */
    repeated string packages = 3;
}

message ShellSubscription {
    repeated SimpleAtomMatcher pushed = 1;
    repeated PulledAtomSubscription pulled = 2;
}